### PR TITLE
A9

### PR DESCRIPTION
**Pull request by BernardZhao**
As promised

### CODE REVIEW

File: labs/a9.md [hunk 1]
```diff
@@ -206,34 +208,46 @@ including signatures and certificates for digital identity verification.
 Suppose you are an important public entity (maybe your pseudonym is Natoshi
```
Comment (64bitpandas) [inline labs/a9.md:208]:
```In [lab a4](a4#generating-and-using-ssh-keys), one task was to generate an SSH
key, using the command:
```
This link is broken

**Review by 64bitpandas — APPROVED** [review]
Did a super quick (probably incomplete) runthrough, everything checks out.

gpg keygeneration seems to be broken for me on tsunami though- running `gpg --full-generate-key` gives the error `gpg: agent_genkey failed: Required environment variable not set
Key generation failed: Required environment variable not set`. I don't know if this also happens on the studentvm's as well.

File: labs/a9.md [hunk 2]
```diff
@@ -290,10 +304,10 @@ Here's an example of hash functions at work. Enter the following commands on
 your student VM (i.e. `ssh username.decal.xcf.sh`) to calculate the SHA1[^sha1]
 hash of a 100MB file.
```
Comment (64bitpandas) [inline labs/a9.md:305]: The actual image seems to be 40mb

**Review by 64bitpandas — APPROVED** [review]

**Comment by BernardZhao** [comment]
> gpg keygeneration seems to be broken for me on tsunami though- running `gpg --full-generate-key` gives the error `gpg: agent_genkey failed: Required environment variable not set Key generation failed: Required environment variable not set`. I don't know if this also happens on the studentvm's as well.

Just tried this on staff3, seems to work

File: labs/a9.md (hunk 1 above)
Comment (BernardZhao) [reply labs/a9.md:208]: I just tried this link and it seems to work for me

**Pull request merged**